detox/src/devices/runtime/drivers/android/AndroidDriver.js

```diff
@@ -394,6 +410,43 @@ class AndroidDriver extends DeviceDriverBase {
       'Press any key to continue...'
     );
   }
+
+  _getLocalBinaryHash(binaryFile) {
+    return generateHash(binaryFile);
+  }
+
+  async _saveHashToRemote(hash, bundleId) {
+    await this.hashHelper.saveHashToRemote(this.adbName, bundleId, hash);
+  }
+
+  async _shouldClearData(bundleId, hash, isPkgInstalled) {
+    const isSameVersionInstalled = this._compareRemoteToLocal(bundleId, hash);
+    return isPkgInstalled && isSameVersionInstalled;
+  }
+
+  async _compareRemoteToLocal(bundleId, localHash) {
+    const hashFilename = `${bundleId}.hash`;
+    const destinationPath = path.posix.join(this.fileXfer.getFilePath(), hashFilename);
```

Comment:
This doesn't make any sense (`this.fileXfer.getFilePath`). `fileXfer` is from type `FileXfer` in your code.